earthly-entrypoint.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/sh
+
+set -e
+
+# Apply global configuration
+if [ -n "$GLOBAL_CONFIG" ]; then
+  earthly config global "$GLOBAL_CONFIG"
+fi
+
+# Apply git configuration
+if [ -n "$GIT_CONFIG" ]; then
+  earthly config git "$GIT_CONFIG"
+fi
+
+# We prefer you use EARTHLY_TOKEN directly, but this is here just in case
+if [ -n "$EARTHLY_EMAIL" ] && [ -n "$EARTHLY_PASSWORD" ]; then
+  earthly account login --email "$EARTHLY_EMAIL" --password "$EARTHLY_PASSWORD"
+fi
+
+# Skip docker if you are not exporting any images.
+if [ -z "$NO_DOCKER" ]; then
+  # Right now, this container is BYOD - Bring Your Own Docker.
+  if [ -z "$DOCKER_HOST" ]; then
+    echo "DOCKER_HOST is not set"
+  fi
+
+  # Light check if docker is functional
+  if ! docker images > /dev/null 2>&1; then
+    echo "Docker appears not to be connected. Please check your DOCKER_HOST variable, and try again."
+    exit 1
+  fi
+fi
+
+# Is container running as privileged? This is currently required.
+if ! captest --text | grep sys_admin > /dev/null; then
```

Comment:
Perhaps theres a better cap to check for? This seemed the most straightforward. Would love to enumerate all the caps we actually need and check for them all...

Reply:
Maybe we shouldn't require privileged if the user brings their own ~bottle~ buildkit?

Reply:
:thinking: Good point. Ill move the check inside.